api/config/service/k8s/v1/k8s.proto

```diff
@@ -8,4 +8,23 @@ import "validate/validate.proto";
 
 message Config {
   repeated string kubeconfigs = 1 [ (validate.rules).repeated = {unique : true} ];
+  RestClientConfig restclientconfig = 2;
```

Comment:
nit: name should be snake case. `kubeconfigs` is one word because `kubeconfig` is a thing but this can be snake case.

```suggestion
  RestClientConfig rest_client_config = 2;
```